spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -777,7 +777,7 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
             int fieldIndex = schema.fieldIndex(field.name());
             Object catalystValue = partition.values().get(fieldIndex, field.dataType());
             Object value = CatalystTypeConverters.convertToScala(catalystValue, field.dataType());
-            values.put(field.name(), value.toString());
+            values.put(field.name(), value == null ? null : value.toString());
```

Comment:
What about `String.valueOf(value)` to avoid the ternary expression?

Reply:
Oh good call. I didn't like the in-line ternary here.

Reply:
`String.valueOf` also ensures that we get the string `"null"` as the value.